### DISCUSSION TITLE
How to trigger Button `onPress` programmatically?

### DISCUSSION THREAD

**Discussion by THOUSAND-SKY** [adobe/react-spectrum#6280] [Q&A]
I have a button using `onPress`, and from non-react code I'd like to trigger the click, but `el.click()` (nor a `ref.current.click()` for that matter) doesn't actually trigger it. Is there a simple way of triggering a press event, like `.click()` would do for `onClick`?

**Comment by snowystinger** [comment]
Is this for a test? If so, I'd suggest looking at our tests.

If this is for an actual interaction, then I'd just call whatever function you're passing to onPress directly. No need to trigger events.

**Reply by THOUSAND-SKY** [reply]
I am trying to click in my browser, via a browser extension and user scripts.

**Reply by snowystinger** [reply]
I see, then you'd probably want to look at our tests for how to trigger them. Or look at userEvent's `click` implementation. Keep in mind, this probably won't result in a trusted user event though.

Depending on how closely tied your app and extension are though, you may find it easier to create direct API's for interfacing between the two.
